libs/blocks/footer/footer.js

```diff
@@ -1 +1,206 @@
+    const regionButton = createTag('a', {
+      class: 'footer-region-button',
+      id: 'region-button',
+      'aria-haspopup': true,
+      'aria-label': regionTextContent,
+      href: '#modal',
+      role: 'button',
+      tabindex: 0,
+    });
+    const regionText = createTag('span', { class: 'footer-region-text' }, regionTextContent);
+    regionButton.insertAdjacentHTML('afterbegin', GLOBE_IMG);
+    regionButton.append(regionText);
+    regionContainer.append(regionButton);
+
+    const regionLinks = region.querySelectorAll('a');
+    regionLinks.forEach((link) => {
+      const selected = link.parentNode.nodeName === 'STRONG';
+      const options = { class: 'footer-region-option' };
+      if (selected) {
+        options.class += ' footer-region-selected';
+        options['aria-current'] = 'page';
+      }
+    });
+    return regionContainer;
+  };
+
+  decorateSocial = () => {
+    const socialEl = this.body.querySelector('.social > div');
+    if (!socialEl) return null;
+    const socialWrapper = createTag('div', { class: 'footer-social' });
+    const socialLinks = createTag('ul', { class: 'footer-social-icons' });
+    socialEl.querySelectorAll('a').forEach((a) => {
+      const domain = a.host.replace(/www./, '').replace(/.com/, '');
+      const supported = ['facebook', 'instagram', 'twitter', 'linkedin'];
+      if (supported.includes(domain)) {
+        const li = createTag('li', { class: 'footer-social-icon' });
+        const socialIcon = createTag('img', {
+          class: 'footer-social-img',
+          loading: 'lazy',
+          src: `/libs/blocks/footer/${domain}-square.svg`,
+          alt: `${domain} logo`,
+          height: '20',
+          width: '20',
+        });
+        a.setAttribute('aria-label', domain);
+        a.textContent = '';
+        a.append(socialIcon);
+        li.append(a);
+        socialLinks.append(li);
+      } else { a.remove(); }
+      socialWrapper.append(socialLinks);
+    });
+    return socialWrapper;
+  };
+
+  decoratePrivacy = () => {
+    const copyrightEl = this.body.querySelector('div em');
+    const links = copyrightEl.parentElement.querySelectorAll('a');
+    if (!copyrightEl || !links) return null;
+    const privacyWrapper = createTag('div', { class: 'footer-privacy' });
+    const copyright = createTag('p', { class: 'footer-privacy-copyright' });
+    copyright.textContent = copyrightEl.textContent;
+    privacyWrapper.append(copyright);
+    const infoLinks = createTag('ul', { class: 'footer-privacy-links' });
+    links.forEach((link) => {
+      const li = createTag('li', { class: 'footer-privacy-link' });
+      if (link.hash === '#interest-based-ads') {
+        link.insertAdjacentHTML('afterbegin', ADCHOICE_IMG);
+      }
+      li.append(link);
+      infoLinks.append(li);
+    });
+    privacyWrapper.append(infoLinks);
+    return privacyWrapper;
+  };
+}
+
+async function fetchFooter(url) {
+  const resp = await fetch(`${url}.plain.html`);
```

Comment:
Not sure if you want to do error handling.  e.g for a 404.

If it is a 404, fetch returns `res.text === '404 Not Found'`
Then on line 198, footerDoc is
```
<html>
  <head></head>
  <body>404 Not Found</body>
</html>
```